lib/src/tokio_util.rs

```diff
@@ -48,6 +49,29 @@ where
         f(&dropper.0)
     })
 }
+
+/// Flatten a nested Result<Result<T>>, defaulting to converting the error type to an `anyhow::Error`.
+/// See https://doc.rust-lang.org/std/result/enum.Result.html#method.flatten
+pub(crate) fn flatten_anyhow<T, E>(r: std::result::Result<Result<T>, E>) -> Result<T>
+where
+    E: Display + Debug + Send + Sync + 'static,
+{
+    match r {
+        Ok(Ok(v)) => Ok(v),
+        Ok(Err(e)) => Err(e),
```

Comment:
Couldn't this be written:

```suggestion
        Ok(x) => x,
```

?

Reply:
Ah nice!  I started to reply "No because..." then I realized if you just look at it, it's clearly true.

Reply:
In my defense this code was inspired by https://github.com/rust-lang/rust/issues/70142#issuecomment-750529061 but they're trying to support a "flatten with" which we're hardcoding to be `anyhow!` effectively.  This code is also not as general because it requires the inner to be an `anyhow::Result` today.